src/Components/AddRoom/AddRoom.tsx

```diff
@@ -0,0 +1,221 @@
+          underline: {
+            '&:after': {
+              borderBottom: 'none',
+            },
+            '&:before': {
+              borderBottom: 'none',
+            },
+          },
+        },
+      },
+    },
+
+  });
+  const VisuallyHiddenInput = styled('input')({
+    clip: 'rect(0 0 0 0)',
+    clipPath: 'inset(50%)',
+    height: 1,
+    overflow: 'hidden',
+    position: 'absolute',
+    bottom: 0,
+    left: 0,
+    whiteSpace: 'nowrap',
+    width: 1,
+  });
+
+  {/*Add Room Function */ }
+
+  const AddNewRoom = (data) => {
+    const formattedSelected = selectedValue.map(({ value }) => value)
+    axios.post('http://154.41.228.234:3000/api/v0/admin/rooms', { ...data, imgs: data.imgs[0], facilities: formattedSelected }, { headers: { ...Headers, "Content-Type": "multipart/form-data" } }).then((response) => {
+      console.log(response);
+    }).catch((error) => {
+      console.log(error);
+
+    })
+
+
+
+  }
+  {/*Get All Facilities */ }
+  const getAllFacilities = () => {
+    axios.get('http://154.41.228.234:3000/api/v0/admin/room-facilities', { headers: Headers }).then((response) => {
+      console.log(response?.data?.data?.facilities);
+      setFacilities(response?.data?.data?.facilities)
+      const newFacilities = response?.data?.data?.facilities
+      const facilities = newFacilities.map(({ _id: value, name: label }) => ({ value, label }))
+      setFacilities(facilities)
+    }).catch((error) => {
+      console.log(error);
+    })
+  }
+
+  React.useEffect(
+    () => { getAllFacilities() }
+    , [])
+  return (
+    <>
+      <Container>
+        <Box sx={{ display: "flex", justifyContent: "center" }}>
+
+          <FormControl component='form' sx={{ display: "flex", justifyContent: "center", alignItems: "center" }} defaultValue="" required onSubmit={handleSubmit(AddNewRoom)}>
+
+            <ThemeProvider theme={theme}>
+
+              <Box sx={{ width: '100%', display: "flex", justifyContent: "center" }}>
+                <Grid container rowSpacing={1} columnSpacing={{ xs: 1, sm: 1, md: 1 }} >
+                  <Grid item xs={12} sx={{ color: "#17202A" }} >
+                    <TextField
+                      InputLabelProps={{
+                        style: { color: 'black' },
+                      }}
+                      sx={{ bgcolor: "#F7F7F7", mb: 5, }}
+                      variant='filled'
+                      label="Room Number"
+                      fullWidth
+                      {...register("roomNumber",{required:true})}
+                    />
+                    {errors.roomNumber&&errors.roomNumber.type==="required"&& <Typography sx={{color:"red"}}>Room Number is required</Typography>}
+
+                  </Grid>
+                </Grid>
+              </Box>
+
+              <Box sx={{ width: '100%' }}>
+                <Grid container rowSpacing={1} columnSpacing={{ xs: 2, sm: 2, md: 2 }}>
+                  <Grid item xs={6} >
+                    <TextField
```

Comment:
The price should be of type number, to avoid users typing letters. and we can ensure also that it is a positive value.